### PR TITLE
[Feature] Pre Authz UI for flow.

### PR DESCRIPTION
**Pull request by moojing**
## Summary

- Add pre-authz  UI for flow.
 

## Related Links

**Asana**:  

**Mockup**:  https://www.figma.com/file/X8Jx6GxcEB9gxoMxujdDSD/7.-Wallet-Web-2.0?node-id=129%3A18156&t=8j6aD96eetC01Y9Q-0

## Checklist

- [x] Pasted Asana or Mockup link.
- [x] Tagged labels.

## Screenshot/Gif

Screenshot 2023-02-10 at 1 03 18 AM

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -53,6 +54,10 @@ const App = () => {
           path={`/:appId/:blockchain(flow)/authz/:authorizationId?`}
           render={() => <Flow.Transaction />}
         />
+        <Route
+          path={`/:appId/:blockchain(flow)/pre-authz`}
+          render={() => <Flow.Transaction />}
+        />
```
Comment (mordochi) [inline src/App.tsx:60]: Why don't we just put it as 
```ts
<Route
   path={`/:appId/:blockchain(flow)/pre-authz`}
   render={() => <Flow.PreAuthz />}
/>
```

File: src/features/Flow/Transaction/index.tsx [hunk 2]
```diff
@@ -81,13 +83,18 @@ const Noop = () => null;
...
     const { setLayoutSize } = useLayoutContext();
 
     // initialization
     useEffect(() => {
-      send({ type: "init", data: state });
+      if (location.pathname.includes("pre-authz")) {
+        send({ type: "preAuthz", data: state });
+      } else {
+        send({ type: "init", data: state });
+      }
```
Comment (mordochi) [inline src/features/Flow/Transaction/index.tsx:97]: I think it'll be easier to figure out that `/pre-authz` actually renders one component specifically if we put it in the `Route` and I supposed that's what `Route`'s purpose is.

File: src/App.tsx (hunk 1 above)
Comment (moojing) [reply src/App.tsx:60]: Do you mean creating another set of components for routing like `src/features/Flow/PreAuthz/index.tsx` / `src/features/Flow/PreAuthz/components/Main.tsx` ?

Comment (moojing) [reply src/App.tsx:60]: I thought of this way before doing this. but we'll have to copy another similar line of code like `Connecting.tsx`. 
But you're right the way routing the PreAuthz is kind of explicit..maybe I should find another way...

Comment (moojing) [reply src/App.tsx]: I've do the change first, at least it's more clear then the previous way, thank you.
https://github.com/portto/wallet-web-ui/pull/42/commits/6c9ebaa141c9a36faf38027e9c40d83b2b064452

Comment (mordochi) [reply src/App.tsx]: thanks!

File: src/features/Flow/PreAuthz/components/Connecting.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      balance,
+    };
+
+    send({
+      type: "ready",
+      data: {
+        user: userData,
+      },
+    });
+  }, [blockchain, send]);
```
Comment (mordochi) [inline src/features/Flow/PreAuthz/components/Connecting.tsx:38]: Looks like we are not using the user's information in the `Main` component, so I guess we don't have to fetch `getUserInfo` and `getAccountAssets` at all.

File: src/features/Flow/NonCustodial/components/Connecting.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      balance,
+    };
+
+    send({
+      type: "ready",
+      data: {
+        user: userData,
+      },
+    });
+  }, [blockchain, send]);
```
Comment (mordochi) [inline src/features/Flow/NonCustodial/components/Connecting.tsx:38]: Same. We only need `authorizationId` and `sessionId` from `context.user` in `Main` so we don't have to fetch from `getUserInfo()` and `getAccountAssets()`.

File: src/features/Flow/PreAuthz/components/Connecting.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import { useCallback, useEffect } from "react";
+import { getAccountAssets, getUserInfo } from "src/apis";
+import Loading from "src/components/Loading";
+import { useTransactionMachine } from "src/machines/transaction";
+import { AccountAsset } from "src/types";
+import fetchDappInfo from "src/utils/fetchDappInfo";
+
+const Connecting = () => {
+  const { context, send } = useTransactionMachine();
```
[the remark this answers is not in the document]
Comment (moojing) [reply src/features/Flow/PreAuthz/components/Connecting.tsx]: Let me try to make it cleaner.
https://github.com/portto/wallet-web-ui/pull/42

File: src/features/Flow/PreAuthz/components/Connecting.tsx (hunk 3 above)
Comment (moojing) [reply src/features/Flow/PreAuthz/components/Connecting.tsx]: remote the code about x-state in the flow of `PreAuthz`
https://github.com/portto/wallet-web-ui/pull/42/commits/210c8aa22d881e1cf999ec782bd2076ae43d4feb

File: src/features/Flow/NonCustodial/components/Connecting.tsx (hunk 4 above)
Comment (moojing) [reply src/features/Flow/NonCustodial/components/Connecting.tsx]: remove the code about x-state int the flow of Flow's NonCustodial.

File: src/features/Flow/NonCustodial/index.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  const [dAppInfo, setdAppInfo] = useState<{ url?: string }>({ url: "" });
+  const sessionId = getItem(KEY_SESSION_ID);
+
+  useEffect(() => {
+    fetchDappInfo({ id: appId }).then((data) => {
+      setdAppInfo(data);
+    });
+  }, [appId]);
+
+  // @todo: fetchDapp info
```
Comment (mordochi) [inline src/features/Flow/NonCustodial/index.tsx:33]: Should we remove this todo? Seems like you already fetch it

File: src/features/Flow/NonCustodial/index.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    if (authorizationId && sessionId) {
+      updateNonCustodial({ authorizationId, sessionId }).then(() => {
+        onTransactionDecline({
+          blockchain,
+          l6n: url,
+          errorMessage: ERROR_MESSAGES.AUTHZ_DECLINE_ERROR,
+        });
+      });
+    }
+  }, [authorizationId, blockchain, sessionId]);
```
Comment (mordochi) [inline src/features/Flow/NonCustodial/index.tsx:45]: Add `dAppInfo` in the dependency

Comment (moojing) [reply src/features/Flow/NonCustodial/index.tsx]: [[Fix]Fix for review.](https://github.com/portto/wallet-web-ui/pull/42/commits/ed95c0a0a0af6e95b96f1e907073a4bef0d48fd1)

File: src/features/Flow/NonCustodial/index.tsx (hunk 6 above)
Comment (moojing) [reply src/features/Flow/NonCustodial/index.tsx]: [[Fix]Fix for review.](https://github.com/portto/wallet-web-ui/pull/42/commits/ed95c0a0a0af6e95b96f1e907073a4bef0d48fd1)

**Review by mordochi — APPROVED** [review]

**Pull request merged**